src/controllers/orderController.js

```diff
@@ -162,4 +164,45 @@ const deleteOrder = async (req, res) => {
   */
 };
 
-module.exports = { createOrder, getAllOrders, getSingleOrder, deleteOrder };
+const handleStripePayment = async (req) => {
+  // Receive Stripe Signature
+  const stripeSignature = req.headers['stripe-signature'];
+
+  //  Construct the Webhook Event
+  const event = stripe.webhooks.constructEvent(
+    req.body, // Raw text body payload received from Stripe
+    stripeSignature, // Value of the `stripe-signature` header from Stripe
+    process.env.STRIPE_CLI_WEBHOOK_SECRET // Webhook Signing Secret
+  );
+
+  //  Retrieve Payment Intent and Order Information
+  const paymentIntent = event.data.object; // Payment intent information
+  const orderId = paymentIntent.metadata.orderId; // Order ID passed in metadata
+  // Additional order-related info if needed
+
+  //  Update Order Status
+  const order = await Order.findById(orderId);
+
+  if (event.type === 'payment_intent.succeeded') {
+    // Update order status to successful
+    order.orderStatus = 'payment_successful';
+  } else if (event.type === 'payment_intent.payment_failed') {
+    // Update order status to failed
+    order.orderStatus = 'payment_failed';
+  }
+
+  // Save the updated order
+  await order.save();
+
+  // Handle other actions based on the event type if needed
+
+  return res.status(200);
```

Comment:
Note the `linter` message here; our function right now only defined `req` as an argument; we should have `res` up on line 167  as well like our other controller functions.